supervisor/process_test.go

```diff
@@ -106,6 +122,91 @@ func TestProcess(t *testing.T) {
 		assertStates(t, p, agent.Status_DONE, agent.Status_STATUS_INVALID)
 	})
 
+	t.Run("Gracefully stop child", func(t *testing.T) {
+		t.Parallel()
+
+		f, err := ioutil.TempFile("", "pmm-agent-process-test-child")
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+		defer func() {
+			require.NoError(t, os.Remove(f.Name()))
+		}()
+
+		t.Logf("building to %s", f.Name())
+		build(t, "child", "process_child.go", f)
+
+		ctx, cancel, l := setup(t)
+		p := newProcess(ctx, &processParams{path: f.Name()}, l)
+
+		assertStates(t, p, agent.Status_STARTING, agent.Status_RUNNING)
+		var logs []string
+		for ; len(logs) == 0; logs = p.Logs() {
+			time.Sleep(50 * time.Millisecond)
+		}
+		cancel()
+		assertStates(t, p, agent.Status_STOPPING, agent.Status_DONE)
+		time.Sleep(200 * time.Millisecond) // Waiting until process is stopped.
```

Comment:
⚠️ `agent.Status_DONE` is returned when a process is stopped. Sleep should not be necessary. If this test fails without it, it is a bug somewhere; we have to find it.

Reply:
yes, test fails without it. I think it takes a few time before system kills children processes.